core/types/receipt_test.go

```diff
@@ -793,85 +800,88 @@ func getOptimismTxReceipts(
-	l2Rcpt := receipts[1]
-	l1GasCost := new(big.Int).Mul(l2Rcpt.L1GasPrice, l2Rcpt.L1GasUsed)
-	l1Fee := new(big.Float).Mul(new(big.Float).SetInt(l1GasCost), l2Rcpt.FeeScalar)
-	require.Equal(t, new(big.Float).SetInt(l2Rcpt.L1Fee), l1Fee)
-}
-
-// This test is based on op-geth
-// https://github.com/ethereum-optimism/op-geth/commit/a290ca164a36c80a8d106d88bd482b6f82220bef
-func TestDeriveOptimismBedrockTxReceipts(t *testing.T) {
-	// Bedrock style l1 attributes with L1Scalar=7_000_000 (becomes 7 after division), L1Overhead=50, L1BaseFee=1000*1e6
-	payload := libcommon.Hex2Bytes("015d8eb900000000000000000000000000000000000000000000000000000000000004d200000000000000000000000000000000000000000000000000000000000004d2000000000000000000000000000000000000000000000000000000003b9aca0000000000000000000000000000000000000000000000000000000000000004d200000000000000000000000000000000000000000000000000000000000004d200000000000000000000000000000000000000000000000000000000000004d2000000000000000000000000000000000000000000000000000000000000003200000000000000000000000000000000000000000000000000000000006acfc0015d8eb900000000000000000000000000000000000000000000000000000000000004d200000000000000000000000000000000000000000000000000000000000004d2000000000000000000000000000000000000000000000000000000003b9aca0000000000000000000000000000000000000000000000000000000000000004d200000000000000000000000000000000000000000000000000000000000004d200000000000000000000000000000000000000000000000000000000000004d2000000000000000000000000000000000000000000000000000000000000003200000000000000000000000000000000000000000000000000000000006acfc0")
-	l1GasPrice := basefee
-	l1GasUsed := bedrockGas
-	feeScalar := big.NewFloat(float64(scalar.Uint64() / 1e6))
-	l1Fee := bedrockFee
-	txs, receipts := getOptimismTxReceipts(t, payload, l1GasPrice, l1GasUsed, feeScalar, l1Fee)
-	senders := []libcommon.Address{libcommon.HexToAddress("0x0"), libcommon.HexToAddress("0x0")}
-
-	// Re-derive receipts.
-	clearComputedFieldsOnReceipts(t, receipts)
-	err := receipts.DeriveFields(params.OptimismTestConfig, blockHash, blockNumber.Uint64(), blockTime, txs, senders)
-	if err != nil {
-		t.Fatalf("DeriveFields(...) = %v, want <nil>", err)
+func TestDeriveOptimismTxReceipts(t *testing.T) {
+	checkBedrockReceipts := func(t *testing.T, receipts, derivedReceipts Receipts) {
+		diffDerivedFields(t, receipts, derivedReceipts)
+		// Check that we preserved the invariant: l1Fee = l1GasPrice * l1GasUsed * l1FeeScalar
+		// but with more difficult int math...
+		l2Rcpt := receipts[1]
+		l1GasCost := new(big.Int).Mul(l2Rcpt.L1GasPrice, l2Rcpt.L1GasUsed)
+		l1Fee := new(big.Float).Mul(new(big.Float).SetInt(l1GasCost), l2Rcpt.FeeScalar)
+		require.Equal(t, new(big.Float).SetInt(l2Rcpt.L1Fee), l1Fee)
+	}
+	checkEcotoneReceipts := func(t *testing.T, receipts, derivedReceipts Receipts) {
+		diffDerivedFields(t, receipts, derivedReceipts)
+		// TODO : add fee calc logic? it already tested in opstack package
```

Comment:
At here, we can check receipts does not hold `l1FeeScalar`. `l1FeeScalar` is not used starting from ecotone. [spec](https://github.com/ethereum-optimism/specs/blob/46130fbccba08354fb292256f178f06716da2f67/specs/protocol/ecotone/l1-attributes.md?plain=1#L80).

Reply:
resolved in [0a84e27](https://github.com/testinprod-io/op-erigon/pull/209/commits/0a84e27fbf6425c34a0a1dd9446673fce4fd8544)